frontend/src/libs/components/input/input.tsx

```diff
@@ -5,35 +5,78 @@ import {
   type FieldValues,
 } from 'react-hook-form';
 
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
 import { useFormController } from '~/libs/hooks/hooks.js';
 
+import styles from './styles.module.scss';
+
 type Properties<T extends FieldValues> = {
   control: Control<T, null>;
   errors: FieldErrors<T>;
-  label: string;
+  label?: string;
   name: FieldPath<T>;
   placeholder?: string;
-  type?: 'text' | 'email' | 'password';
+  type?: 'text' | 'email' | 'password' | 'textarea';
+  isActive?: boolean;
+  isFilled?: boolean;
+  isDisabled?: boolean;
 };
 
 const Input = <T extends FieldValues>({
   control,
   errors,
-  label,
+  label = '',
   name,
   placeholder = '',
   type = 'text',
+  isActive,
+  isFilled,
+  isDisabled,
 }: Properties<T>): JSX.Element => {
   const { field } = useFormController({ name, control });
 
   const error = errors[name]?.message;
   const hasError = Boolean(error);
+  const inputStyles = [
+    styles.input,
+    isActive && styles.active,
+    isFilled && styles.filled,
+    isDisabled && styles.disabled,
+    error && styles.error,
+  ];
+  const Input = (
+    <>
+      <input
+        {...field}
+        type={type}
+        placeholder={placeholder}
+        className={getValidClassNames(...inputStyles)}
+        disabled={isDisabled}
+      />
+      {type === 'password' && (
+        <span className={styles.passwordEye}>&#128065;</span>
+      )}
+    </>
+  );
+
+  const InputArea = (
+    <textarea
+      {...field}
+      placeholder={placeholder}
+      className={getValidClassNames(...inputStyles, styles.textarea)}
+      disabled={isDisabled}
+    />
+  );
 
   return (
-    <label>
+    <label className={styles.label}>
       <span>{label}</span>
```

Comment:
if label is not passed, this element should not be rendered

Reply:
I can replace it, with div in such case, because I use it as wrapper for styles?

Reply:
replaced with div when no label